jupyterlab_bigquery/jupyterlab_bigquery/tests/pagedAPI_handler/pagedQuery_handler_test.py

```diff
@@ -0,0 +1,247 @@
+import unittest
+from unittest.mock import Mock, MagicMock, patch
+from jupyterlab_bigquery.pagedAPI_handler.pagedQuery_handler import PagedQueryHandler
+import json
+from google.cloud import bigquery
+
+
+def multiple_return_helper(return_values):
+
+  def inner():
+    return return_values.pop(0)
+
+  return inner
+
+
+class TestPagedQueryHandler(unittest.TestCase):
+
+  @patch("jupyterlab_bigquery.pagedAPI_handler.PagedAPIHandler.__init__")
+  @patch.object(bigquery, 'Client')
+  def setUp(self, fake_bigquery_client, fake_super):
+    self.client_mock = MagicMock()
+    fake_bigquery_client.return_value = self.client_mock
+
+    self.dummy_query_handler = PagedQueryHandler(None, None)
+
+  def tearDown(self):
+    self.dummy_query_handler.generator_pool.clear()
+
+  def test_ctor_client(self):
+    self.assertEqual(self.dummy_query_handler.client, self.client_mock)
+
+  @patch(
+      'jupyterlab_bigquery.pagedAPI_handler.'+\
+      'pagedQuery_handler.PagedQueryHandler.client.query'
+  )
+  @patch.object(bigquery, 'QueryJobConfig')
+  def test_query_dry(self, fake_job_config, fake_client_query):
+    query = MagicMock()
+    jobConfig = MagicMock()
+    dryRunOnly = True
+    page_size = 2000
+    job_id = 1231
+
+    request_body = {
+        'query': query,
+        'jobConfig': jobConfig,
+        'dryRunOnly': dryRunOnly
+    }
+
+    dry_run_job = MagicMock()
+    dry_run_job.job_id = job_id
+
+    dry_run_job_config_mock = MagicMock()
+
+    fake_client_query.return_value = dry_run_job
+    fake_job_config.return_value = dry_run_job_config_mock
+
+    gen = self.dummy_query_handler.query(request_body, page_size)
+
+    actual_dry_run_job, actual_job_id = next(gen)
+
+    fake_job_config.assert_called_with(dry_run=True, use_query_cache=False)
+    fake_client_query.assert_called_with(query,
+                                         job_config=dry_run_job_config_mock)
+    self.assertEqual(actual_dry_run_job, dry_run_job)
```

Comment:
I generally prefer not to compare to a Mock object unless the data types are from an external library that is difficult to reproduce.

Is it possible to compare actual_dry_run_job to a real object instead of a mock. Or just assert that actual_dry_run_job.job_id == dry_run_job.job_id.